modules/electrophysiology_browser/php/file_reader.class.inc

```diff
@@ -38,12 +38,16 @@ class File_Reader extends \NDB_Page
             $filename = urldecode(basename($file));
             $path     = dirname($file);
 
-            $downloader = new \LORIS\FilesDownloadHandler(
-                new \SPLFileInfo($downloadpath . $path)
-            );
-            return $downloader->handle(
-                $request->withAttribute('filename', $filename)
-            );
+            try {
+                $downloader = new \LORIS\FilesDownloadHandler(
+                    new \SPLFileInfo($downloadpath . $path)
+                );
+                return $downloader->handle(
+                    $request->withAttribute('filename', $filename)
+                );
+            } catch (\Exception $e) {
+                return new \LORIS\Http\Response\JSON\NotFound();
+            }
```

Comment:
This has the chance of masking other errors than 404

I suggest 1 of 2 approaches:
1. check that the directory/file exists first above this code and return a 404 if it's not dowloadable ahead of calling the FilesDownloadHandler
2. in you catch, catch a specific exception that says its not found and then have a generic catch for other errors? this one might be more complicated depending on what error FilesDownloadHandler throws